Nuxt 3 build error with `@unocss/nuxt` in ~0.48.3

## Description

Hi, upgrading the `@unocss/nuxt` module from 0.46.5 to ~0.48.3 introduces a build error.

I don't have time to track it down. If you need more information, please let me know.

<details>
<summary>Error log</summary>

```
❯ pnpm build

> package@0.0.2 build ***
> nuxt build

Nuxi 3.0.0 
Nuxt 3.0.0 with Nitro 1.0.0

i Client built in 3786ms
i Building server...

 ERROR  [unocss:global:build:scan] [postcss] ***\server.mjs-unocss-hash.css:1:3544: Double colon


 ERROR  [postcss] ***\server.mjs-unocss-hash.css:1:3544: Double colon

  at Input.error (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\input.js:148:16)
  at Parser.doubleColon (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parser.js:561:22)
  at Parser.colon (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parser.js:516:16)
  at Parser.checkMissedSemicolon (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parser.js:581:22)
  at Parser.decl (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parser.js:279:12)
  at Parser.other (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parser.js:128:18)
  at Parser.parse (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parser.js:72:16)
  at parse (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\parse.js:11:12)
  at new LazyResult (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\lazy-result.js:133:16)
  at Processor.process (***\node_modules\.pnpm\postcss@8.4.20\node_modules\postcss\lib\processor.js:28:14)

***:
ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  ***@0.0.2 build: `nuxt build`
Exit status 1
```

</details>

The dev server works as expected.

## Steps to reproduce

Reproduction: https://stackblitz.com/edit/nuxt-3-unocss

Manually run build in the terminal: `pnpm build`.

Possibly related: #1965

### Conditions

1. update `@unocss/nuxt` to ~0.48.3, the exact version is unknown
2. an element with an attribute of `style="display: none"` in an SFC
3. a script block with some content (i.e. non-empty, I just put `//` in there) in the same SFC

If any of the conditions are not met, the error may not occur.

--- a/packages/preset-attributify/src/extractor.ts
+++ b/packages/preset-attributify/src/extractor.ts
@@ -58,7 +58,7 @@ export const extractorAttributify = (options?: AttributifyOptions): Extractor =>
             const extractTernary = Array.from(content.matchAll(/(?:[\?:].*?)(["'])([^\1]*?)\1/gms))
               .map(([,,v]) => v.split(splitterRE)).flat()
             return (extractTernary.length ? extractTernary : content.split(splitterRE))
-              .filter(Boolean)
+              .filter(v => Boolean(v) && v !== ':')
               .map(v => `[${name}~="${v}"]`)
           }
         })

